src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -130,6 +137,11 @@ public async Task Invoke(HttpContext context)
             return;
         }
 
+        if (authenticateResult != null && !authenticateResult.Succeeded)
+        {
+            _logger.LogDebug("Policy authentication schemes {policyName} did not succeed", String.Join(", ", policy.AuthenticationSchemes));
```

Comment:
What about logging this in the policyEvaluator instead?

Reply:
That would be also an option.
I just wanted as few log entries as possible. For me, it would have been enough to know that the combination does not work instead of each individual policy.